code/bsg.c

```diff
@@ -1,9 +1,24 @@
-#include "bsg.h"
 #include <stdio.h>
+#include "bsg.h"
 
-StateBSG selectBSGMode(Vehicle vehicle, Engine engine, BSG bsg,
-                       Battery battery) {
-  return BSG_INDLE;
+StateBSG selectBSGMode(Vehicle vehicle, Engine engine, BSG bsg, Battery battery) {
+   if (vehicle.requestCarStart != 0) {
+      return BSG_STARTER;
+  };
+  if (vehicle.velocity == 0) {
+    return BSG_IDLE;
+  };
+  if (vehicle.angleBrakePedal > 0) {
+      return BSG_GENERATOR;
+  };
+  if (vehicle.angleBrakePedal == 0){
+    return BSG_STARTER;
+  }
+  if (battery.voltage >= 0) {
+      return BSG_STARTER;
+  };
+  if (battery.current >= 0) {
+      return BSG_STARTER;
+  };
```

Comment:
Suggestion:  Group all similar ifs using bitwise or operator instead using multiple ifs.

The following code would generate the same behavior:

```suggestion
  if ( (vehicle.angleBrakePedal == 0) || (battery.voltage >= 0) || (battery.current >= 0) ) {
      return BSG_STARTER;
  };
```